utils/genUnicodeTable.py

```diff
@@ -132,6 +227,846 @@ def print_categories(unicode_data_lines):
+            get_canonical_name=lambda fields: fields[1] if fields[0] == "gc" else None,
+        )
+        gc_property_ranges = parse_codepoint_ranges(
+            UnicodeDataFiles.get_lines("DerivedGeneralCategory.txt"),
+            lambda canonical_name: canonical_name in gc_property_aliases.keys(),
+        )
+
+        pool = self.general_category_pool
+
+        # Update the string pool with the General_Category property names and
+        # aliases.
+        for name, aliases in gc_property_aliases.items():
+            pool.add_aliases(name, aliases)
+
+        # These General_Category properties are never directly associated with
+        # codepoints, but exist conceptually as unions of other properties.
+        #
+        # NOTE: It's important that any ranges shared by compound groups overlap,
+        #       so that the offset+size can be contiguous for each of them.
+        #
+        # <https://www.unicode.org/reports/tr44/#General_Category_Values>
+        COMPOUND_GC_PROPERTIES = {
+            "C": ["Cc", "Cf", "Cn", "Co", "Cs"],
+            "L": ["Ll", "Lt", "Lu", "Lm", "Lo"],
+            "LC": ["Ll", "Lt", "Lu"],
+            "M": ["Mc", "Me", "Mn"],
+            "N": ["Nd", "Nl", "No"],
+            "P": ["Pc", "Pd", "Pe", "Pf", "Pi", "Po", "Ps"],
+            "S": ["Sc", "Sk", "Sm", "So"],
+            "Z": ["Zl", "Zp", "Zs"],
+        }
+
+        cat = "General_Category"
+        for compound_name, canonical_names in COMPOUND_GC_PROPERTIES.items():
+            pool.mark_range_pool(cat, compound_name)
+
+            for canonical_name in canonical_names:
+                ranges = gc_property_ranges[canonical_name]
+                if pool.get_range(cat, canonical_name) is None:
+                    pool.mark_range_pool(cat, canonical_name, ranges)
+
+            pool.mark_range_array_pool(compound_name, canonical_names)
+
+        # Add any extra ranges that are not part of the compound groups.
+        for canonical_name, ranges in gc_property_ranges.items():
+            if pool.get_range(cat, canonical_name) is None:
+                pool.mark_range_pool(cat, canonical_name, ranges)
+                pool.mark_range_array_pool(canonical_name, [canonical_name])
+
+    def gather_binary_properties(self):
+        """
+        Gather allowed binary (in the true/false sense) property aliases and
+        codepoint ranges, explicitly given by ECMA262, into the shared string
+        and range pools.
+
+        <https://tc39.es/ecma262/multipage/text-processing.html#table-binary-unicode-properties>
+
+        Example property aliases input:
+
+            # Alias ; Canonical name  ; Additional alias
+
+            AHex    ; ASCII_Hex_Digit
+            Alpha   ; Alphabetic
+            WSpace  ; White_Space     ; space
+        """
+        binary_property_aliases = {
+            canonical_name: []
+            for canonical_name in "ASCII ASCII_Hex_Digit Alphabetic Bidi_Control Bidi_Mirrored "
```

Comment:
I think it would be preferable to just make this a list to begin with instead of creating and splitting a string. It can also be assigned to a separate variable beforehand to make the loop easier to read.

Reply:
No problem, I've gone ahead and made that change.